tests/services/phase.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+
+let phaseService: any;
+let githubService: any;
+let userService: any;
+
+describe('PhaseService', () => {
+    beforeEach(() => {
+      githubService = jasmine.createSpyObj('GithubService',
+        ['fetchSettingsFile', 'storePhaseDetails', 'createRepository']);
+      userService = new UserService(null, null);
+      phaseService = new PhaseService(null, githubService, null, userService, null);
+    });
+
+    describe('.fetchSessionData()', () => {
+      it('should fetch the correct settings file', () => {
+        githubService.fetchSettingsFile.and.returnValue(of(mockSettingsFile));
+        phaseService.fetchSessionData().subscribe((sessionData: SessionData) => {
+          expect(sessionData).toEqual(mockSessionData);
+        });
+      });
+    });
+
+    describe('.updateSessionParameters()', () => {
+      it('should update the currentPhase and repoName based on given sessionData', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+
+        expect(phaseService.currentPhase).toEqual(Phase.phaseBugReporting);
+        expect(phaseService.repoName).toEqual(mockSettingsFile[Phase.phaseBugReporting]);
+      });
+    });
+
+    describe('.attemptSessionAvailabilityFix()', () => {
+      it('should throw no errors given phaseBugReporting and student role', () => {
+        userService.currentUser = testStudent;
+        // refresh phaseService with new userService
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).not.toThrowError();
+      });
+
+      it('should throw an error given no openPhases', () => {
+        userService.currentUser = testStudent;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Current Phase\'s Repository has not been opened.')
+        );
+      });
+
+      it('should throw an error given non-student role and phaseBugReporting', () => {
+        userService.currentUser = testTutor;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Bug-Reporting Phase\'s repository initialisation is only available to Students.')
+        );
```

Comment:
Feels quite uncomfortable to expect a specific string in the error since changing the error message slightly would cause this test to fail 😅

Do you think it would be better to to just leave it at `.toThrow()` without defining the specific error in which case it would just check that an error is expected no matter what the message is? Since we are mocking the `createRepository` from GHService, the only area for an error to be thrown would be inside `attemptSessionAvailabilityFix` itself?

So something like this?

```suggestion
        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow();
```

Reply:
Yes, I think that would be better to check if it just throws an error in case we change the error message!